### DISCUSSION TITLE
企业微信消息接收推送疑问

### DISCUSSION THREAD

**Discussion by zero1389498** [w7corp/easywechat#2828] [General]
EasyWeChat文档 接口
例如 企业微信客服消息 $app->kf_message->sync(string $cursor, string $token, int $limit);  传参是三个
![image](https://github.com/user-attachments/assets/dc2cbdf3-4ea3-41a8-9383-a761eb322523)

但是官方企业微信 的接口 请求为 https://qyapi.weixin.qq.com/cgi-bin/kf/sync_msg?access_token=ACCESS_TOKEN 
但是企业官方文档 需要再url传 
```
{
    "cursor": "4gw7MepFLfgF2VC5npN",
    "token": "ENCApHxnGDNAVNY4AaSJKj4Tb5mwsEMzxhFmHVGcra996NR",
    "limit": 1000,
    "voice_format": 0,
    "open_kfid": "wkxxxxxx"
}
```
对比之下,包的 ?access_token=ACCESS_TOKEN 是没有吗?
5.x版本 php 7.4

**Comment by overtrue** [comment]
中间件里

**Reply by zero1389498** [reply]
大佬,看一下我下面的回复

**Comment by zero1389498** [comment]
调用读取信息接口参数的异常
`
文档写法
$app->kf_message->sync(string $cursor, string $token, int $limit);
`
`
//我的写法
$data = $app->kf_message->sync('', $message['Token'], 1);
`
回调提示 无效 open_kfid | 官方文档  95000=不合法的open_kfid
`
{"errcode":95000,"errmsg":"invalid open_kfid, hint: [1722483383315352538205003], from ip: 159.75.204.160, more info at https:\/\/open.work.weixin.qq.com\/devtool\/query?e=95000","msg_list":[]}
`

然后我改了 sync 接口传参 如下
![image](https://github.com/user-attachments/assets/cd39554f-7b9c-45c7-b0ce-80e28a68cedd)

MessageClient 类下的 sync 代码 修改如下 

![image](https://github.com/user-attachments/assets/6a800f00-2142-4fb4-8fec-cefd2546e111)

回调的信息就正常了 回调数据如下

`{"errcode":0,"errmsg":"ok","next_cursor":"4gw7MepFLfgF2VC","msg_list":[{"msgid":"4KqXJ3SgFEALDSPxkzREzmi3mVkCa6Yw","send_time":1722218397,"origin":4,"msgtype":"event","event":{"event_type":"enter_session","scene":"","open_kfid":"wkTpEeDQAaCled5U1o4XZBHw","external_userid":"wmTm-IxEo5bQ_UZ6Ar7KAjxjQ","welcome_code":"AH_Mfn0SgOCfZOwTFLn7aZSLsWLJ2Y"}}],"has_more":1}`

版本 为 "overtrue/wechat": "~5.0"
